src/tuning/tune_hyperparameters.py

```
@@ -0,0 +1,361 @@
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+    stream=sys.stdout,  # Log to console
+)
+logger = logging.getLogger(__name__)
+
+# --- Helper Functions ---
+
+
+def generate_hyperparameter_combinations() -> Iterator[Dict[str, Any]]:
+    """Generate all hyperparameter combinations from the search space."""
+    keys, values = zip(*SEARCH_SPACE.items())
```

Ruff (0.12.2)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)